### PR TITLE
Fix `AttributeError`s in tic-tac-toe command

### LINKED ISSUE #772: Handle the AI in Tic-Tac-Toe requester check.
## Description
image

The `!ttt` command currently has a `is_requester_free` check, which ensures that players cannot run more than one game at a time. To reproduce, run `.ttt` and then while the game is still running, run `.ttt` again.

However, the check does not check for games played against the `AI` class when getting the Discord user for a player in each game, resulting in an AttributeError.

## Would you like to implement a fix?

- [ ] I'd like to implement the bug fix
- [X] Anyone can implement the bug fix

### LINKED ISSUE #898: AttributeError: 'AI' object has no attribute 'user'
Sentry Issue: [SIR-LANCEBOT-6M](https://sentry.io/organizations/python-discord/issues/2704303843/?referrer=github_integration)

```
AttributeError: 'AI' object has no attribute 'user'
(8 additional frame(s) were not displayed)
...
  File "discord/ext/commands/core.py", line 1124, in can_run
    return await discord.utils.async_all(predicate(ctx) for predicate in predicates)  # type: ignore
  File "discord/utils.py", line 522, in async_all
    elem = await elem
  File "bot/exts/fun/tic_tac_toe.py", line 240, in predicate
    return all(
  File "bot/exts/fun/tic_tac_toe.py", line 241, in <genexpr>
    ctx.author not in (player.user for player in game.players) for game in ctx.cog.games if not game.over
  File "bot/exts/fun/tic_tac_toe.py", line 241, in <genexpr>
    ctx.author not in (player.user for player in game.players) for game in ctx.cog.games if not game.over

Unhandled command error: 'AI' object has no attribute 'user'
```

### PR DESCRIPTION
**Pull request by TizzySaurus**
## Relevant Issues
Closes #898.
Closes #772.

## Description
- Resolved the `AI.user` AttributeError by setting it to the bot (`ctx.me`)
- Resolved the `Game.channel` AttributeError by setting it to `ctx.channel`.

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

**Comment by Xithrius** [comment]
image

This is the error that occurs now when two users try to play against AI, one after the other, while one game is still in progress. Is this the desired output?

**Comment by TizzySaurus** [comment]
> image
> 
> This is the error that occurs now when two users try to play against AI, one after the other, while one game is still in progress. Is this the desired output?

The code from before this PR made it so that their can only even be one command per channel, and the user invoking the command cannot already be playing a game.

I guess the missing `channel` attribute of the `Game` class prevented this from actually working, which is why the error only now shows.

TL;DR; This is the behaviour the original writer intended, but if we want it changed I can do so.
